apollo-router/src/ageing_priority_queue.rs

```diff
@@ -0,0 +1,124 @@
+use std::sync::atomic::AtomicUsize;
```

Comment:
It would be great to see some unit tests for this new module.

Reply:
Added some